tests/functional/modules/test_zos_archive_func.py

```diff
@@ -0,0 +1,935 @@
+# limitations under the License.
+
+
+from __future__ import absolute_import, division, print_function
+import time
+
+import pytest
+__metaclass__ = type
+
+SHELL_EXECUTABLE = "/bin/sh"
+USS_TEMP_DIR = "/tmp/archive"
+USS_TEST_FILES = {  f"{USS_TEMP_DIR}/foo.txt" : "foo sample content",
+                    f"{USS_TEMP_DIR}/bar.txt": "bar sample content", 
+                    f"{USS_TEMP_DIR}/empty.txt":""}
+USS_EXCLUSION_FILE = f"{USS_TEMP_DIR}/foo.txt"
+TEST_PS = "USER.PRIVATE.TESTDS"
+TEST_PDS = "USER.PRIVATE.TESTPDS"
+HLQ = "USER"
+MVS_DEST_ARCHIVE = "USER.PRIVATE.ARCHIVE"
+
+USS_DEST_ARCHIVE = "testarchive.dzp"
+
+STATE_ABSENT = 'absent'
+STATE_ARCHIVED = 'archive'
+STATE_COMPRESSED = 'compress'
+STATE_INCOMPLETE = 'incomplete'
+
+USS_FORMATS = ['tar', 'zip', 'gz', 'bz2', 'pax']
+
+def set_uss_test_env(ansible_zos_module, test_files):
+    for key, value in test_files.items():
+        ansible_zos_module.all.shell(
+            cmd=f"echo \"{value}\" > \"{key}\"",
+            executable=SHELL_EXECUTABLE,
+        )
+
+def create_multiple_data_sets(ansible_zos_module, base_name, n, type, ):
+    test_data_sets = []
+    for i in range(n):
+        curr_ds = dict(name=base_name+str(i),
+                       type=type,
+                       state="present",
+                       replace=True,
+                       force=True)
+        test_data_sets.append(curr_ds)
+
+        
+    # Create data sets in batch
+    ansible_zos_module.all.zos_data_set(
+        batch=test_data_sets
+    )
+    return test_data_sets
+
+def create_multiple_members(ansible_zos_module, pds_name, member_base_name, n):
+    test_members = []
+    for i in range(n):
+        curr_ds = dict(name="{0}({1})".format(pds_name, member_base_name+str(i)),
+                       type="member",
+                       state="present",
+                       replace=True,
+                       force=True)
+        test_members.append(curr_ds)
+    ansible_zos_module.all.zos_data_set(
+        batch=test_members
+    )
+    return test_members
+
+######################################################
+#
+# USS TEST
+#
+######################################################
+"""
+List of tests:
+- test_uss_single_archive
+- test_uss_single_archive_with_mode
+- test_uss_single_archive_with_force_option
+- test_uss_archive_multiple_files
+- test_uss_archive_multiple_files_with_exclude_path
+- test_uss_archive_remove_targets
+"""
+
+
+# Core functionality tests
+# Test archive with no options
+@pytest.mark.uss
+@pytest.mark.parametrize("format", USS_FORMATS)
+def test_uss_single_archive(ansible_zos_module, format):
+    try:
+        hosts = ansible_zos_module
+        hosts.all.file(path=f"{USS_TEMP_DIR}", state="absent")
+        hosts.all.file(path=USS_TEMP_DIR, state="directory")
+        set_uss_test_env(hosts, USS_TEST_FILES)
+        dest = f"{USS_TEMP_DIR}/archive.{format}"
+        expected_state = STATE_ARCHIVED
```

Comment:
If is already declared the assertion should be only 
`assert result.get("dest_state") == STATE_ARCHIVED `

Reply:
agree